egs/wsj/s5/steps/nnet3/get_egs.sh

```diff
@@ -257,11 +273,14 @@ if [ $stage -le 2 ]; then
     copy-int-vector ark:- ark,scp:$dir/ali.ark,$dir/ali.scp || exit 1;
 fi
 
-egs_opts="--left-context=$left_context --right-context=$right_context --compress=$compress"
+[ $ivector_interval -gt $frames_per_eg ] && echo "Error ivector_interval should not be greater than frames_per_eg." && exit 1
+echo $ivector_interval > $dir/info/ivector_interval
+
+egs_opts="--left-context=$left_context --right-context=$right_context --compress=$compress --left-shift-window=$left_shift_window --ivector-interval=$ivector_interval"
 
 [ -z $valid_left_context ] &&  valid_left_context=$left_context;
 [ -z $valid_right_context ] &&  valid_right_context=$right_context;
-valid_egs_opts="--left-context=$valid_left_context --right-context=$valid_right_context --compress=$compress"
+valid_egs_opts="--left-context=$valid_left_context --right-context=$valid_right_context --compress=$compress --left-shift-window=$left_shift_window --ivector-interval=$ivector_interval"
 
```

Comment:
Do we assume that the model is provided to get_egs.sh ?
Or are you talking about extracting the ivector-period during decoding, when we do have the model available ?